### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by filnet**
- [ ] Tested on all platforms changed
- [X] Compilation warnings were addressed
- [X] `cargo fmt` has been run on this branch
- [ ] `cargo doc` builds successfully
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by filnet** [comment]
Would it make sense to fail the builds if there are warnings like the on this PR fixes ?

**Comment by madsmtm** [comment]
Thanks for this, these have been bothering me for a while!

> Would it make sense to fail the builds if there are warnings like the ones this PR fixes ?

I would say yes (can easily be done with `RUSTFLAGS: "-D warnings"`)

**Comment by filnet** [comment]
> I would say yes (can easily be done with `RUSTFLAGS: "-D warnings"`)

Where would one do that. There are no CI related files in the winit repo.

**Comment by maroider** [comment]
> There are no CI related files in the winit repo.

GitHub actions looks in `.github/workflows` for workflows, and our CI is in `.github/workflows/ci.yml`.

**Comment by filnet** [comment]
The wasm build fail as expected.

**Comment by filnet** [comment]
I am getting spammed by CI messages about unrelated failed checks whenever I push changes to the branch.
Doesn't make for a good user experience when contributing to winit.

The failed checks should be disabled until properly fixed ?

**Comment by filnet** [comment]
> GitHub actions looks in `.github/workflows` for workflows, and our CI is in `.github/workflows/ci.yml`.

Totally overlooked them ! Thanks.

**Comment by madsmtm** [comment]
Please don't disable the failing `nightly` checks; they're very useful for spotting future incompatibility errors (see #2078), and they don't prevent people from merging the PR. If we disable them, we forget them :wink:

**Comment by maroider** [comment]
> and they don't prevent people from merging the PR

They don't??

Now that you've mentioned it, the "Squash and merge" button does look awfully clickable.

**Comment by madsmtm** [comment]
> The notification spam is unfortunate, but I don't know if we can turn those off.

I don't think we can, though people can control it themselves under [Settings => Notifications => Actions](https://github.com/settings/notifications).

**Comment by filnet** [comment]
Ping.

File: examples/request_redraw_threaded.rs [hunk 1]
```diff
@@ -1,6 +1,9 @@
+#[cfg(not(target_arch = "wasm32"))]
 use std::{thread, time};
 
+#[cfg(not(target_arch = "wasm32"))]
 use simple_logger::SimpleLogger;
+#[cfg(not(target_arch = "wasm32"))]
```
Comment (maroider) [inline examples/request_redraw_threaded.rs:6]: This doesn't look all that great, but it's not that big of an issue either.

**Review by maroider — APPROVED** [review]

File: examples/request_redraw_threaded.rs (hunk 1 above)
Comment (maroider) [reply examples/request_redraw_threaded.rs:6]: > I agree but what's the alternative ?

Realistically, there's the option of moving the imports into the `main` function. There's also the option of using the `xtask` pattern to work around this, but that's more work.

Comment (filnet) [reply examples/request_redraw_threaded.rs]: I moved the imports in the `main` function.

**Comment by rukai** [comment]
Ah, just a heads up that warnings are not stable.
So if you want to fail on warnings you should really pin your rust version, see https://github.com/rukai/mado_rust_library_ci_template/ for more details.

**Comment by madsmtm** [comment]
Well, we run CI on `nightly`, which would surface changes to warnings that would start failing our CI in a way that prevents a merge.

I'm not strictly opposed to pinning our Rust version, though I think it should be for other reasons like guaranteeing a MSRV.

**Comment by rukai** [comment]
Ah sure, if your paying enough attention to the nightly workflow then that would work too.

**Pull request merged**